karamel-ui/src/main/java/se/kth/karamel/webservice/KaramelServiceApplication.java

```diff
@@ -845,18 +842,138 @@ public Response sudoPassword(SudoPasswordJSON sudoPwd) {
+        }
+
+        return response;
+      }
+
+    }
+
+    @Path("/setGithubCredentials")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public static class SetGithubCredentials {
+
+      @POST
+      public Response setGithubCredentials(@FormParam("user") String user, @FormParam("password") String password) {
+        Response response = null;
+        Logger.getLogger(KaramelServiceApplication.class.getName()).
+            log(Level.INFO, " Received request to set github credentials.... ");
+        try {
+          GithubUser githubUser = karamelApiHandler.registerGithubAccount(user, password);
+
+          response = Response.status(Response.Status.OK).
+              entity(githubUser).build();
+        } catch (KaramelException e) {
+          e.printStackTrace();
+          response = Response.status(Response.Status.INTERNAL_SERVER_ERROR).
+              entity(new StatusResponseJSON(StatusResponseJSON.ERROR_STRING, e.getMessage())).build();
+        }
+
+        return response;
+      }
+    }
+
+    @Path("/getGithubOrgs")
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class GetGithubOrgs {
+
+      @POST
+      public Response getGithubOrgs() {
+        Response response = null;
+        Logger.getLogger(KaramelServiceApplication.class.getName()).
+            log(Level.INFO, " Received request to set github credentials.... ");
+        try {
+          List<OrgItem> orgs = karamelApiHandler.listGithubOrganizations();
+          response = Response.status(Response.Status.OK).
+              entity(orgs).build();
+        } catch (KaramelException e) {
+          e.printStackTrace();
+          response = Response.status(Response.Status.INTERNAL_SERVER_ERROR).
+              entity(new StatusResponseJSON(StatusResponseJSON.ERROR_STRING, e.getMessage())).build();
+        }
+
+        return response;
+      }
+    }
+
+    @Path("/getGithubRepos")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public static class GetGithubRepos {
+
+      @POST
+      public Response getGithubRepos(@FormParam("org") String org) {
+        Response response = null;
+        Logger.getLogger(KaramelServiceApplication.class.getName()).
+            log(Level.INFO, " Received request to set github credentials.... ");
+        try {
+          List<RepoItem> repos = karamelApiHandler.listGithubRepos(org);
+          response = Response.status(Response.Status.OK).
+              entity(repos).build();
+        } catch (KaramelException e) {
+          e.printStackTrace();
+          response = Response.status(Response.Status.INTERNAL_SERVER_ERROR).
+              entity(new StatusResponseJSON(StatusResponseJSON.ERROR_STRING, e.getMessage())).build();
+        }
+
+        return response;
+      }
+    }
+
+//    @Path("/createGithubRepo")
```

Comment:
Ok, i misunderstood your comment. If you write 'comment', it can be misunderstood as a command - "comment on this - write comments". If you wrote "commented out" or even "comments", then i would have got it. I started out with a wider API, and progressively narrowed it down to "create, add, push and commit" in a single operation.